infra/repo_manager.py

```diff
@@ -202,7 +202,8 @@ def checkout_commit(self, commit):
     utils.execute(['git', 'checkout', '-f', commit],
                   self.repo_dir,
                   check_result=True)
-    utils.execute(['git', 'clean', '-fxd'], self.repo_dir, check_result=True)
+    if clean:
+      utils.execute(['git', 'clean', '-fxd'], self.repo_dir, check_result=True)
```

Comment:
nit: maybe add a comment like the one you posted on this PR, would be helpful for future reference